packages/component-library/src/components/form/mt-link/mt-link.vue

```diff
@@ -0,0 +1,171 @@
+<template>
+  <component
+    :is="elementType"
+    class="mt-link"
+    :class="linkClasses"
+    :aria-disabled="disabled"
+    :tabindex="disabled ? -1 : 0"
+    v-bind="to ? { ...$attrs, to } : $attrs"
+    @click="onClick"
+  >
+    <slot />
+  </component>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+import type { PropType } from "vue";
+
+export default defineComponent({
+  name: "MtLink",
+
+  props: {
+    to: {
+      type: String,
+      required: false,
+      default: undefined,
+    },
+    size: {
+      type: String,
+      required: false,
+      default: "default",
+      validator(value: string) {
+        if (!value.length) {
+          return true;
+        }
+        return ["small", "default", "large"].includes(value);
+      },
+    },
+    variant: {
+      type: String as PropType<"primary" | "secondary" | "critical">,
+      required: false,
+      default: "primary",
+      validator(value: string) {
+        if (!value.length) {
+          return true;
+        }
+        return ["primary", "secondary", "critical"].includes(value);
+      },
+    },
+    disabled: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    underline: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
```

Comment:
Do we have cases where we don't want an underline? Refering again to the Figma component. But I could be wrong and we may have many cases where we don't have an underline.

Reply:
I guess there will be some cases, like links by product name to open a details page in a table, or some design ideas where the link will not have an underline. It depends on the design concept. 
However, it's also possible that it's just an edge case, I removed it, users can use it themselves with the style they want if they need to.